core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -642,6 +642,19 @@ class SnappyParser(session: SnappySession)
         ) |
         MATCH ~> UnresolvedAttribute.quoted _
     ) |
+    '{' ~ FN ~ ws ~ identifier ~ '(' ~ (expression * commaSep) ~ ')' ~ ws ~ '}' ~ ws ~> {
+      (f: Any, e: Any) =>
+        f.asInstanceOf[String] match {
+          case f if f.equalsIgnoreCase("TIMESTAMPADD") =>
```

Comment:
@soubhik-c This is not proper. We cannot clutter the parser with such special case handling and set such an example in code (else others will also follow suit). Please add a separate TimestampAdd function instead which will handle SQL_TSI_DAY (and more later if required) and internally call DateAdd as required (or extend latter).

The one and only case of special handling here is for COUNT because that is the only one which can accept COUNT(*) else even that should have been avoided if possible.